src/bibgetter.py

```diff
@@ -341,17 +506,177 @@ def format(filename):
         stdout=subprocess.DEVNULL,
     )
 
+def substitute_bibtex_key(entry_text, expected_key):
```

Comment:
Is this function necessary, given the `IDS` field? The philosophy behind bibgetter is that it's 2025, people should just use `biblatex`.

That being said: maybe it makes sense to have a `--bibtex` option that returns `bibtex` (rather than `biblatex`) output.

Reply:
"The philosophy behind bibgetter is that it's 2025, people should just use biblatex" -- maybe! But the editor tooling is an issue. At least my emacs+AucTeX+reftex config does not autocomplete extra IDs from the "ids" list (to be precise: when I type a prefix of an ID from the "ids" list, autocomplete matches the record, but accepting the autocompletion erases the prefix and replaces it with the record ID). At least the way I set it up (many years ago) works like that. And I don't like this since it makes the TeX source less readable. I'm not sure how many editors support the alternative IDs fully.

Reply:
Ah yes, I can see why this can be useful to others (I just swear by MathSciNet identifier, if possible, and arXiv if not).

Then I think that the implementation makes sense as-is: you return a version of the entry with the preferred key, and `ids` contains the alternative keys, suitable adapted to the preferred key being used.